README.md

```diff
@@ -18,4 +22,44 @@ custom:
         localEnvVarStages:
             - local
             - test
+    functions:
+        calculate:
+            handler: src/handlers.calculate
+            environment:
+                # BEWARE: This will require KMS decryption to be useable
+                MY_SECRET_VAR: ${self:custom.myVaultVar}
+                # This will be plaintext
+                MY_CONFIG_VAR: ${self:custom.myConsulVar}
+            events:
+                - http:
+                path: /v1/calculate
+                method: post
+```
+
+## Upgrade to plugin 1.0 / slsv3:
+Serverless v3 contained some [https://www.serverless.com/framework/docs/guides/upgrading-v3](major changes) to how custom variables and parameters are used, notably that arbitrary CLI parameters are no longer permitted. WealthWizards used these heavily. Thankfully they also added in params as a good alternative. When migrating a project to serverless framework v3 / v1.0 of this plugin, you will need be mindful of the following:
+
+- Any use of custom variables dependant on `opt` interpolation and custom cli vars will no longer work, e.g.
+```yaml
+  tenant: ${opt:tenant, 'tenant'}
+```
+This should be replaced by the use of params:
+```yaml
+  tenant: ${param:tenant}
+```
+Params can be specified on the CLI, and helpfully you can also provide per stage params in your yml alongside a default. As a simple example:
+```yaml
+params:
+  default:
+    accountId: xxxx
+    accountName: proof
+  red:
+    accountId: yyyy
+    accountName: prod
 ```
+
+- Stage is the source of truth for your environment. Remove any redundant variables such as `{opt.environment}`, these should no longer be required when using `params`. This makes deploying personal stacks and feature branches simpler too, as `params` allows for having default parameter sets within YML (see above).
+
+_Aside_: From what I can glean, this does mean that compiling a single artifact of code + CF template is no longer possible, as the stage and its parameters are required at _package time_. These are then baked into the Cloudformation template.
+
+- Bonus breakage: Changed interface and usage to be more direct. `serviceConfig` and `secretConfig` have been replaced by `consul` and `vault` respectively. The existing pattern of using `secretConfig` pointing to a Consul path, whose value contains a Vault path, has been replaced with a direct vault path.
```

Comment:
Just for my understanding: going directly to the vault path, I'm assuming that's because we're moving away from using `app_config_vars`? It was why it was done this way in the first place.

Reply:
So it's very specific to our use of `app_config_vars` and at the time we did this, the idea was to make the serverless deployments the same experience as for the kubernetes services i.e: add/update the definition of the service in `app_config_vars` including version, config and secrets and use `konan` to deploy.

Config and secrets (as in vault paths) values end up in Consul from `app_config_vars`. So `serviceConfig` was used to get the Config values out of Consul and `secretConfig` was used to retrieve vault paths from Consul first, then get the actual secret from Vault and then encrypt it again with KMS.

It was really to shift the config from the Serverless yaml config to `app_config_vars` which was used heavily then.

Hope this makes sense 😅